test/internal/throws.js

```diff
@@ -1,11 +1,14 @@
 'use strict';
 
-var assert = require('assert');
+var assert = require ('assert');
 
-//  throws :: (Function, TypeRep a, String) -> Undefined
-module.exports = function throws(f, type, message) {
-  assert.strictEqual(arguments.length, throws.length);
-  assert.throws(f, function(err) {
-    return err.constructor === type && err.message === message;
-  });
+var equals = require ('./equals');
+
+//  throws :: (() -> Undefined !) -> Error -> Undefined !
+module.exports = function throws(thunk) {
+  assert.strictEqual (arguments.length, throws.length);
+  return function throws$1(expected) {
+    assert.strictEqual (arguments.length, throws$1.length);
+    assert.throws (thunk, equals (expected));
+  };
 };
```

Comment:
The internal `throws` function is now curried, and now takes a single value representing the expected error rather than a type representative and an error message.

Before:

```javascript
throws(thunk, TypeError, 'Invalid value');
```

After:

```javascript
throws (thunk) (new TypeError ('InvalidValue'));
```